Tasks/BashV3/bash.ts

```diff
@@ -72,7 +73,20 @@ async function run() {
                 targetFilePath = input_filePath;
             }
 
-            contents = `. '${targetFilePath.replace("'", "'\\''")}' ${input_arguments}`.trim();
+            if (old_source_behavior) {
+                contents = `. '${targetFilePath.replace("'", "'\\''")}' ${input_arguments}`.trim();
+            } else {
+                // Check if executable bit is set
+                let stats: fs.Stats = tl.stats(input_filePath);
+                if ((stats.mode & 1) > 0) {
+                    contents = `bash '${targetFilePath.replace("'", "'\\''")}' ${input_arguments}`.trim();
+                }
+                else {
+                    console.log(stats.mode);
```

Comment:
Do we want this to be a debug log and have some description of what it's printing?

Reply:
Good catch.